### PR TITLE
Add support for $CARGO_HOME in servobuild

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

File: python/servo/command_base.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ class CommandBase(object):
     def __init__(self, context):
         self.context = context
 
+        def resolverelative(category, key):
+            if self.config[category][key].startswith("./"):
```
Comment (SimonSapin) [inline python/servo/command_base.py]: `path.join` already ignore arguments before one that is an absolute path: `path.join('/foo', 'bar') == '/foo/bar', path.join('/foo', '/bar') == '/bar'`. Why special-casing `./`?

Comment (Manishearth) [inline python/servo/command_base.py]: We also want to handle `~`.

I'll handle `~` replacement first, then do `path.join`.

Comment (SimonSapin) [inline python/servo/command_base.py]: You want to call [`os.path.expanduser`](https://docs.python.org/2/library/os.path.html#os.path.expanduser) first.

Comment (Manishearth) [inline python/servo/command_base.py]: fixed

File: python/servo/command_base.py [hunk 2]
```diff
@@ -51,6 +51,14 @@ class CommandBase(object):
     def __init__(self, context):
         self.context = context
 
+        def resolverelative(category, key):
+            # Allow ~
+            self.config[category][key] = path.expanduser(self.config[category][key])
+            # Resolve relative paths
+            self.config[category][key] = path.join(context.topdir,
+                                                   self.config[category][key])
+            print self.config[category][key]
```
Comment (SimonSapin) [inline python/servo/command_base.py]: Is this for debugging this change?

Comment (Manishearth) [inline python/servo/command_base.py]: Yes, fixed

**Comment by SimonSapin** [comment]
Looks good.

**Pull request closed without merging**